src/NodaTime.Test/DurationTest.Construction.cs

```diff
@@ -120,10 +120,20 @@ public void FromTicks(long ticks)
         }
 
         [Test]
+        [TestCase(Duration.MinDays, Duration.MinDays, 0)]
         [TestCase(1.5, 1, NanosecondsPerDay / 2)]
         [TestCase(-0.25, -1, 3 * NanosecondsPerDay / 4)]
         [TestCase(100000.5, 100000, NanosecondsPerDay / 2)]
         [TestCase(-5000, -5000, 0)]
+        [TestCase(-0.1, -1, 9 * NanosecondsPerDay / 10)]
+        [TestCase(0.1, 0, NanosecondsPerDay / 10)]
+        [TestCase(0.01, 0, NanosecondsPerDay / 100)]
+        [TestCase(4.1, 4, NanosecondsPerDay / 10)]
```

Comment:
I found that `Duration.FromDays(4.1)` actually worked correctly with the old logic, unlike with the other methods. That's why I had tested with `2.53` on my PR instead, as it was a value I found that did cause rounding issues. Up to you if you want to change or add that as another case

Reply:
Ah, great - yes, will add that as another test.